catalog/core/catalog-email-api-impl/src/main/java/org/codice/alliance/core/email/impl/EmailSenderImpl.java

```diff
@@ -0,0 +1,186 @@
+     */
+    @Override
+    public void sendEmail(String fromEmail, String toEmail, String subject, String body,
+            List<Pair<String, InputStream>> attachments) throws IOException {
+        notNull(fromEmail, "fromEmail must be non-null");
+        notNull(toEmail, "toEmail must be non-null");
+        notNull(subject, "subject must be non-null");
+        notNull(body, "body must be non-null");
+        notNull(attachments, "attachments must be non-null");
+
+        if (StringUtils.isBlank(mailHost)) {
+            throw new IOException("the mail server hostname has not been configured");
+        }
+
+        List<File> tempFiles = new LinkedList<>();
+
+        try {
+            InternetAddress emailAddr = new InternetAddress(toEmail);
+            emailAddr.validate();
+
+            Properties properties = createSessionProperies();
+
+            Session session = Session.getDefaultInstance(properties);
+
+            MimeMessage mimeMessage = new MimeMessage(session);
+            mimeMessage.setFrom(new InternetAddress(fromEmail));
+            mimeMessage.addRecipient(Message.RecipientType.TO, emailAddr);
+            mimeMessage.setSubject(subject);
+
+            BodyPart messageBodyPart = new MimeBodyPart();
+            messageBodyPart.setText(body);
+
+            Multipart multipart = new MimeMultipart();
+            multipart.addBodyPart(messageBodyPart);
+
+            Holder<Long> bytesWritten = new Holder<>(0L);
+
+            for (Pair<String, InputStream> attachment : attachments) {
+
+                messageBodyPart = new MimeBodyPart();
+                File file = File.createTempFile("email-sender-", ".dat");
+                tempFiles.add(file);
+
+                copyDataToTempFile(file, attachment.getValue(), bytesWritten);
+                messageBodyPart.setDataHandler(new DataHandler(new FileDataSource(file)));
+                messageBodyPart.setFileName(file.getAbsolutePath());
+                multipart.addBodyPart(messageBodyPart);
+
+            }
+
+            mimeMessage.setContent(multipart);
+
+            Transport.send(mimeMessage);
+
+            LOGGER.debug("Email sent to " + toEmail);
+
+        } catch (AddressException e) {
+            throw new IOException("invalid email address: email=" + toEmail, e);
+        } catch (MessagingException e) {
+            throw new IOException("message error occured on send", e);
+        } finally {
+            tempFiles.forEach(file -> {
+                if (!file.delete()) {
+                    LOGGER.debug("unable to delete tmp file: path=" + file);
+                }
+            });
+        }
+    }
+
+    private void copyDataToTempFile(File file, InputStream inputStream, Holder<Long> bytesWritten)
+            throws IOException {
+
+        try (OutputStream outputStream = new FileOutputStream(file)) {
+
+            byte[] buffer = new byte[4096];
+
+            int c;
+            while ((c = inputStream.read(buffer)) != -1) {
```

Comment:
the idea was to quickly fail if the size of the attachments exceed the attachment limit. I could use IOUtils and check the size after each attachment is added, but that could still be a really large amount of data buffered to disk (and fill the disk)